### PR TITLE
Update: End-to-end setup instruction of SigmaDAO app

### PR DESCRIPTION
**Pull request by rmeena840**

### CODE REVIEW

File: dapp/README.md [hunk 1]
```diff
@@ -32,33 +32,88 @@ Add proposal page of SigmaDAO. On this page user can add a proposal on a existin
 
 ## Setup
 
-Install the dependencies
+This section demonstrates end-to-end setup required to run SigmaDAO application full functioning.
```
Comment (rmeena840) [inline dapp/README.md]:
```suggestion
This section demonstrates the end-to-end setup required to run the SigmaDAO application fully functioning.
```

File: dapp/README.md [hunk 2]
```diff
@@ -32,33 +32,88 @@ Add proposal page of SigmaDAO. On this page user can add a proposal on a existin
...
+```
+Example of sigmadao-indexer running on private-net:
+
+```bash
+cmd/algorand-indexer/algorand-indexer daemon --data-dir /tmp -P "host=localhost port=5432 user=algorand password=indexer dbname=pgdb sslmode=disable" --algod-net 127.0.0.1:4001 --algod-token aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa --genesis ../algo-builder/infrastructure/node_data/genesis.json
+```
+
+### How to create a sigmada user and setup sigmadao with read-only permission to database?
+
+Setup the sigma_daos table. Sigmadao-indexer should be running before executing below scripts.
```
Comment (rmeena840) [inline dapp/README.md]:
```suggestion
Sigmadao-indexer should be running before executing the below scripts.
```

File: dapp/README.md [hunk 3]
```diff
@@ -32,33 +32,88 @@ Add proposal page of SigmaDAO. On this page user can add a proposal on a existin
...
+
+### How to create a sigmada user and setup sigmadao with read-only permission to database?
+
+Sigmadao-indexer should be running before executing the below scripts.
+
+Create a sigma dao user, if not already created:
+
+    make create-sigma-dao-user
+
+Setup sigma dao, indexer should be running before executing this script:
```
Comment (rmeena840) [inline dapp/README.md]:
```suggestion
Setup SigmaDAO with indexer:
```

File: dapp/README.md [hunk 4]
```diff
@@ -32,33 +32,88 @@ Add proposal page of SigmaDAO. On this page user can add a proposal on a existin
...
+
+Run below script to delete the sigma dao user, sigma dao user should be present:
+
+    make drop-sigma-dao-user
+
+### How to setup sigmadao backend?
+
+Make sure you are in /dapp directory of this project before executing below steps.
+
+Start the development server on http://localhost:4000 configured for private-net.
```
Comment (rmeena840) [inline dapp/README.md]:
```suggestion
Start the backend server on http://localhost:4000 configured for private-net.
```

File: dapp/README.md [hunk 5]
```diff
@@ -32,33 +32,88 @@ Add proposal page of SigmaDAO. On this page user can add a proposal on a existin
...
+Make sure you are in /dapp directory of this project before executing below steps.
+
+Start the backend server on http://localhost:4000 configured for private-net.
+
+```bash
+yarn server:privatenet
 ```
 
-## Setup the Sigma Dao role and table in PostgreSQL with read only access to database
+Note: The above backend server is running configured for private-net network. You can configure it to MainNet, TestNet or BetaNet using below command:
```
Comment (rmeena840) [inline dapp/README.md]:
```suggestion
Note: The above backend server is configured for private-net network. You can configure it to MainNet, TestNet or BetaNet using the below command:
```

File: dapp/README.md [hunk 6]
```diff
@@ -32,33 +32,88 @@ Add proposal page of SigmaDAO. On this page user can add a proposal on a existin
...
 
-## Setup the Sigma Dao role and table in PostgreSQL with read only access to database
+Note: The above backend server is configured for private-net network. You can configure it to MainNet, TestNet or BetaNet using the below command:
+
+```bash
+yarn server:[mainnet/testnet/betanet]
+```
 
-Follow the examples/dao [README.MD](https://github.com/scale-it/algo-builder/blob/develop/examples/dao/README.md#setup-sigma-dao) in algo-builder to setup the Sigma Dao role and table in PostgreSQL.
+Once, abobe backend setup is done. The backend should be running on [localhost](http://localhost:4000) which connects to PostgreSQL database. [Postgraphile](https://www.graphile.org/postgraphile/) has automatic resolver which reads the schema of table and function for auto query generation.
```
Comment (rmeena840) [inline dapp/README.md]:
```suggestion
Once, the above backend setup is done. The backend should be running on [localhost](http://localhost:4000) which connects to PostgreSQL database. [Postgraphile](https://www.graphile.org/postgraphile/) has automatic resolver which reads the schema of table and function for auto query generation.
```

File: dapp/README.md [hunk 7]
```diff
@@ -32,33 +32,88 @@ Add proposal page of SigmaDAO. On this page user can add a proposal on a existin
...
 
 ```bash
-yarn install
+make
+```
+4. Please make sure postgres server is running.
+5. Sigmadao-indexer can be run on any network using below command:
+
+```bash
+cmd/algorand-indexer/algorand-indexer daemon --data-dir /tmp -P "host=localhost port=5432 user=<postgres-user> password=<postgres-password> dbname=<postgres-database-name> sslmode=disable" --algod-net <link-to-network> --algod-token <network-token> --genesis <path-to-genesis-file>
```
Comment (vuvoth) [inline dapp/README.md:54]: What is cmd/algorand-indexer? this is path?

Comment (rmeena840) [reply dapp/README.md:54]: It's a command directory where many commands are generated.

File: dapp/README.md [hunk 8]
```diff
@@ -32,33 +32,88 @@ Add proposal page of SigmaDAO. On this page user can add a proposal on a existin
...
+
+    make setup-sigma-dao
+
+Run below script to delete the sigma dao user, sigma dao user should be present:
+
+    make drop-sigma-dao-user
+
+### How to setup sigmadao backend?
+
+Make sure you are in /dapp directory of this project before executing below steps.
```
Comment (Megha-Dev-19) [inline dapp/README.md]:
```suggestion
Make sure you are in `/dapp` directory of this project before executing below steps.
```

**Review by Megha-Dev-19 — COMMENTED** [review]
Shouldn't we add the steps on how to deploy the contract.? As that is the 1st step to do, and the reason behind adding the `/contracts` folder

**Comment by rmeena840** [comment]
> Shouldn't we add the steps on how to deploy the contract.? As that is the 1st step to do, and the reason behind adding the `/contracts` folder

@Megha-Dev-19 The goal is to add end-to-end steps required to make SigmaDAO application fully working. SigmaDAO just uses /contract folder for smart contract. We specifically don't need steps for smart contract to get SigmaDAO working fully.

**Comment by Megha-Dev-19** [comment]
>  We specifically don't need steps for smart contract to get SigmaDAO working fully.

But how would the user gets the information from the server if no contract is deployed.? @robert-zaremba what do you think about it?

**Comment by rmeena840** [comment]
> > We specifically don't need steps for smart contract to get SigmaDAO working fully.
> 
> But how would the user gets the information from the server if no contract is deployed.? @robert-zaremba what do you think about it?
@Megha-Dev-19 
The user will be able to deploy the contract only if the user has set up the SigmaDAO, indexer, and backend. There is no setup required for smart contract.

**Review by Megha-Dev-19 — APPROVED** [review]

**Pull request merged**